File: array/153.go
func findMin(nums []int) int {
    
    if len(nums) == 0 {
        return 0
    }
    
    var high, low, mid int
    high = len(nums) - 1
    //find the pivot, pivot will be the smallest value as well
    for low < high {
        mid = (low + high) / 2
        if nums[mid] > nums[high] {
            //异常区段
            low = mid + 1
        }else {
            high = mid
        }
    }
    return nums[low]
}

File: binary_search/153.cc
class Solution {
public:
    int findMin(vector<int>& nums) {
        int start = 0, end = nums.size()-1;
        int mid = (start + end) / 2;
        while (start < end) {
            if (nums[mid] > nums[end]) {
                if (nums[mid] > nums[mid+1])
                    return nums[mid+1];
                start = mid+1;
            } else {
                end = mid;
            }
            mid = (start + end) / 2;
        }
        if (start == end)
            return nums[0];
        return nums[start];
    }
};
